src/sbsign/sbsign.c

```diff
@@ -344,33 +240,231 @@ static int verb_sign(int argc, char *argv[], void *userdata) {
                 return log_error_errno(SYNTHETIC_ERRNO(EIO), "Failed to convert SpcIndirectDataContent to BER: %s",
                                        ERR_error_string(ERR_get_error(), NULL));
 
+        *ret_idc = TAKE_PTR(idcraw);
+        *ret_idcsz = (size_t) idcrawsz;
+
+        return 0;
+}
+
+static int asn1_timestamp(ASN1_TIME **ret) {
+        ASN1_TIME *time;
+        uint64_t epoch = UINT64_MAX;
+        int r;
+
+        assert(ret);
+
+        r = secure_getenv_uint64("SOURCE_DATE_EPOCH", &epoch);
+        if (r != -ENXIO)
+                log_debug_errno(r, "Failed to parse $SOURCE_DATE_EPOCH, ignoring: %m");
+
+        if (epoch == UINT64_MAX) {
+                time = X509_gmtime_adj(NULL, 0);
+                if (!time)
+                        return log_error_errno(SYNTHETIC_ERRNO(EIO), "Failed to get current time: %s",
+                                               ERR_error_string(ERR_get_error(), NULL));
+        } else {
+                time = ASN1_TIME_set(NULL, (time_t) (epoch / USEC_PER_SEC));
+                if (!time)
+                        return log_oom();
+        }
+
+        *ret = TAKE_PTR(time);
+
+        return 0;
+}
+
+static int pkcs7_configure(X509 *certificate, EVP_PKEY *private_key, PKCS7 **ret_p7, PKCS7_SIGNER_INFO **ret_si) {
+        int r;
+
+        assert(certificate);
+        assert(private_key);
+        assert(ret_p7);
+        assert(ret_si);
+
```

Comment:
some comments what these functions do would be very good. openssl is opaque enough, would be really good to give the occasional reader a bit of hint what this does because "configure" is not a very explanatory moniker...